sql/src/test/java/org/apache/druid/sql/calcite/schema/SystemSchemaTest.java

```diff
@@ -330,10 +330,10 @@ public void testGetTableMap()
   public void testSegmentsTable() throws Exception
   {
     // total segments = 6
-    // segments 1,2,3 are published and available
-    // segments 4,5,6  are published but unavailable
-    // segment 3 is published but not served
-    // segment 2 is served by 2 servers, so num_replicas=2
+    // segments 1,2,3 are published but unavailable
+    // segments 4,5,6  are published and available
```

Comment:
Something about this comment doesn't sound right. Segments _can be_ served by historicals but not be published. This happens when a segment has been marked unused in the metadata store, but hasn't been dropped from historicals yet. In the case of delays in dropping segments (for some reason: slowness, bugs, etc) this condition could stick around for a while, and in that case, we want the system table to reflect reality accurately to aid in diagnosis.

So please add tests for this case too.

Reply:
> Segments can be served by historicals but not be published. This happens when a segment has been marked unused in the metadata store, but hasn't been dropped from historicals yet.

Ah, nice catch, didn't think about that case as well.

Reply:
Good points, so i added segment `test1` , `test2` to metadata response, but left out `test3`, which tests the case @gianm pointed out. `test3` is not published, but is available from historical.

Reply:
Thanks. Part of the goal of system tables is to help people diagnose issues when things don't quite "add up". So it's important that the fields mean what they are supposed to, rather than inferring them from other things (since those inferences might be invalid if the system is in a funky state).